### PR TITLE
[StructApp] Bushing element

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
This PR implements `BushingElement` which couples displacement and rotational dofs and can have either constant stiffness or tabulated force/displacement and moment/rotation data in each direction if required.

**🆕 Changelog**
- Added `BushingElement`
- Added tests

### CODE REVIEW

**Comment by philbucher** [comment]
Nice, I would like to take a look
If I havent replied by Monday evening then please go ahead without my review ;)

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.cpp [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+    // ALL MEMBER VARIABLES THAT MUST BE CLONED HAVE TO BE DEFINED HERE
+
+    BushingElement NewElement(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+
+    return Kratos::make_intrusive<BushingElement>(NewElement);
+}
+
+BushingElement::~BushingElement()
+{
+}
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/bushing_element.cpp:118]: unused

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.h [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        KRATOS_CLASS_POINTER_DEFINITION(NonLinearStiffness);
+
+        NonLinearStiffness(
+            const Variable<double>& rVariableX,
+            const Variable<double>& rVariableY)
+            : mpVariableX(&rVariableX),
+              mpVariableY(&rVariableY)
+        {};
+
+        ~NonLinearStiffness() = default;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/bushing_element.h]:
```suggestion
```

better let the compiler do its job ;)

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.h [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    };
+
+    class ConstantStiffness: public Stiffness
+    {
+    public:
+        KRATOS_CLASS_POINTER_DEFINITION(ConstantStiffness);
+
+        ConstantStiffness(const Variable<double>& rVariable) : mpVariable(&rVariable) {};
+
+        ~ConstantStiffness() = default;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/bushing_element.h]: same

```suggestion
```

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.h [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+        Variable<double> const * mpVariableX;
+
+        Variable<double> const * mpVariableY;
+    };
+
+    ///@}
+    ///@name Private member variables
+    ///@{
+
+    array_1d<Stiffness::UniquePointer, 6> mStiffnessGetters;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/bushing_element.h:292]: why not std array?

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.h [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+// Project includes
+#include "includes/element.h"
+#include "includes/ublas_interface.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) BushingElement: public Element
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/bushing_element.h:27]: can you please add some docs? It is not very clear what this does/how it should be used

**Review by philbucher — DISMISSED** [review]
nice

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.h (hunk 4 above)
Comment (sunethwarna) [reply applications/StructuralMechanicsApplication/custom_elements/bushing_element.h:292]: no specific reason... I can change it :)

File: applications/StructuralMechanicsApplication/custom_elements/bushing_element.h (hunk 5 above)
Comment (sunethwarna) [reply applications/StructuralMechanicsApplication/custom_elements/bushing_element.h:27]: I will :)

**Review by philbucher — APPROVED** [review]

**Comment by sunethwarna** [comment]
The documentation is being added in the #12771. Hence merging this :)

**Pull request merged**